custom_components/adaptive_lighting/switch.py

```diff
@@ -463,6 +468,23 @@ def color_difference_redmean(
     blue_term = (2 + (255 - r_hat) / 256) * delta_b**2
     return math.sqrt(red_term + green_term + blue_term)
 
+def check_direction_change(
+    last: int,
+    current: int,
+    last_adapt_value: int 
+) -> bool:
+    if last_adapt_value < last: # brightness adapting down.
+        _LOGGER.debug("compare direction: current %s larger than last %s", current, last)
+        if current > last:
+            return True
+        if current < last_adapt_value:
+            return True
+    elif last_adapt_value > last: # brightness adapting up.
+        if current < last:
+            return True
+        if current > last_adapt_value:
+            return True
+    return False
```

Comment:
I believe this is the same
```
if last_adapt_value < last: # Brightness adapting down
    return current > last or current < last_adapt_value
elif last_adapt_value > last: # Brightness adapting up
    return current < last or current > last_adapt_value
return False
```

Reply:
Absolutely correct, those exist in that format because during testing I had _LOGGER.debug on every check